x-pack/plugins/security_solution/public/flyout/document_details/right/components/assignees.tsx

```diff
@@ -0,0 +1,131 @@
+import type { AssigneesIdsSelection } from '../../../../common/components/assignees/types';
+import { AssigneesPopover } from '../../../../common/components/assignees/assignees_popover';
+import { UsersAvatarsPanel } from '../../../../common/components/user_profiles/users_avatars_panel';
+import { useSetAlertAssignees } from '../../../../common/components/toolbar/bulk_actions/use_set_alert_assignees';
+import { ASSIGNEES_ADD_BUTTON_TEST_ID, ASSIGNEES_TITLE_TEST_ID } from './test_ids';
+
+const UpdateAssigneesButton: FC<{ togglePopover: () => void }> = memo(({ togglePopover }) => (
+  <EuiToolTip
+    position="left"
+    content={i18n.translate(
+      'xpack.securitySolution.flyout.right.visualizations.assignees.popoverTooltip',
+      {
+        defaultMessage: 'Assignees',
+      }
+    )}
+  >
+    <EuiButtonIcon
+      aria-label="Update assignees"
+      data-test-subj={ASSIGNEES_ADD_BUTTON_TEST_ID}
+      iconType={'plusInCircle'}
+      onClick={togglePopover}
+    />
+  </EuiToolTip>
+));
+UpdateAssigneesButton.displayName = 'UpdateAssigneesButton';
+
+export interface AssigneesProps {
+  /**
+   * Id of the document
+   */
+  eventId: string;
+
+  /**
+   * The array of ids of the users assigned to the alert
+   */
+  assignedUserIds: string[];
+
+  /**
+   * Callback to handle the successful assignees update
+   */
+  onAssigneesUpdated?: () => void;
+}
+
+/**
+ * Document assignees details displayed in flyout right section header
+ */
+export const Assignees: FC<AssigneesProps> = memo(
+  ({ eventId, assignedUserIds, onAssigneesUpdated }) => {
+    const setAlertAssignees = useSetAlertAssignees();
+
+    const uids = useMemo(() => new Set(assignedUserIds), [assignedUserIds]);
+    const { data: assignedUsers } = useBulkGetUserProfiles({ uids });
+
+    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+    const onSuccess = useCallback(() => {
+      if (onAssigneesUpdated) onAssigneesUpdated();
+    }, [onAssigneesUpdated]);
+
+    const togglePopover = useCallback(() => {
+      setIsPopoverOpen((value) => !value);
+    }, []);
+
+    const onAssigneesApply = useCallback(
+      async (assigneesIds: AssigneesIdsSelection[]) => {
+        setIsPopoverOpen(false);
+        if (setAlertAssignees) {
+          const updatedIds = removeNoAssigneesSelection(assigneesIds);
+          const assigneesToAddArray = updatedIds.filter((uid) => !assignedUserIds.includes(uid));
+          const assigneesToRemoveArray = assignedUserIds.filter((uid) => !updatedIds.includes(uid));
+
+          const assigneesToUpdate = {
+            assignees_to_add: assigneesToAddArray,
+            assignees_to_remove: assigneesToRemoveArray,
+          };
+
+          await setAlertAssignees(assigneesToUpdate, [eventId], onSuccess, noop);
+        }
+      },
+      [assignedUserIds, eventId, onSuccess, setAlertAssignees]
+    );
+
+    return (
+      <EuiFlexGroup alignItems="center" direction="row" gutterSize="xs">
```

Comment:
you will want to add `responsive={false}` here to avoid having a weird look in small screens
Normal screen size
![Screenshot 2023-11-07 at 3 01 19 PM](https://github.com/elastic/kibana/assets/17276605/b8d3e33e-d375-49bb-b8a2-49d55adf301e)

Small Screen
![Screenshot 2023-11-07 at 3 01 26 PM](https://github.com/elastic/kibana/assets/17276605/e9db5b8d-ff18-4aba-8569-f5a53ee441e8)

I have the same changes to the rest of the header in another PR, waiting to be reviewed